packages/react/src/utils/partition/partition.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Partitions an `array` in two with elements that satisfy the `predicate` then
+ * those that don't, respectively.
+ *
+ * @param array List to be partitioned.
+ * @param predicate Function that determines in where to put each element.
+ */
+export const partition = <T>(
```

Comment:
Do you see this being used elsewhere rather than for `partitionObject`?

Reply:
Not really, maybe for something like separating tokens perhaps

```ts
partition([...colors, ...borderRadii], (token) => token.startsWith('ods-color-'));
```

Reply:
In that case maybe it could go inside `partitionObject.js` file?

It's a little confusing, to me a utility is something that can be (and will be) used in multiple places.

Separating props is a such utility. But maybe if these two are merged into one file, and final utility is called `partitionProps`?

What do you think?

Reply:
It's a bit preparing for the future I suppose.
The only thing I don't like about `partitionProps` is that it can work on any object, not just props.
I don't mind removing partition for now though, we can always reintroduce later when needed.